frontend/src/component/incomingWebhooks/IncomingWebhooksModal/IncomingWebhooksForm/useIncomingWebhooksForm.ts

```diff
@@ -0,0 +1,136 @@
+import { useIncomingWebhooks } from 'hooks/api/getters/useIncomingWebhooks/useIncomingWebhooks';
+import { IIncomingWebhook } from 'interfaces/incomingWebhook';
+import { useEffect, useState } from 'react';
+
+const INCOMING_WEBHOOK_NAME_REGEX = /^[A-Za-z0-9\-_]*$/;
```

Comment:
Do we have this validation in the backend? It's nice we do these validations in the frontend but we should also have them in the backend. In general, I don't like to repeat validations both in frontend and backend, but this gives a quick response to the user. I'd consider server validation over frontend validation.

Reply:
We do have it, shared 1on1. 

The server validation is actually less strict than this one, since it only checks for url safeness. Surprisingly, some less common characters are considered URL safe. 

This one is stricter, only checking for alphanumeric characters, `-` and `_`, which seemed a bit more sane to me. I like the fact that the UI _suggests_ this set of characters while the server technically accepts anything that is URL safe, but we can make them act the same way if you think that's a better approach.

Reply:
Regarding repeating them, this feels saner to me than pinging the server in _real-time_ every time the value changes.